### PR TITLE
Implement Forewarn, Frisk, and BattlerTags for Magnet Rise (WIP)

### PR DESCRIPTION
**Pull request by TempsRay**
* Implemented Forewarn and created OHKO move flags
* Implemented Frisk to check the opponent's abilities rather than their items since all items are already visible
* Created BattlerTags for Magnet Rise implementation but have not implemented it yet since that would result in some changes to Smack Down and Thousand Arrows too

### CODE REVIEW

**Review by Flashfyre — DISMISSED** [review]

File: src/data/move.ts [hunk 1]
```diff
@@ -69,7 +69,8 @@ export enum MoveFlags {
   DANCE_MOVE = 4096,
   WIND_MOVE = 8192,
   TRIAGE_MOVE = 16384,
-  IGNORE_ABILITIES = 32768
+  IGNORE_ABILITIES = 32768,
+  OHKO_Move = 65536
```
[the remark this answers is not in the document]
Comment (TempsRay) [reply src/data/move.ts:73]: Is there a quick and easy way to check if a move is a OHKO move? Then again, I could just hardcode it in the forewarn method considering I already did so for counter/mirror coat/metal burst

[the remark this answers is not in the document]
Comment (lucfd) [reply src/data/move.ts:73]: You can use Move's findAttr() method and then check if `OneHitKOAttr` is present. Something like this: `move.getMove().findAttr(attr => attr instanceof OneHitKOAttr)`

File: src/data/ability.ts [hunk 2]
```diff
@@ -1685,6 +1685,53 @@ function getAnticipationCondition(): AbAttrCondition {
...
+    let maxPowerSeen = 0;
+    let maxMove = "";
+    let movePower = 0;
+    for (let opponent of pokemon.getOpponents()) {
+      for (let move of opponent.moveset) {
+        if (move.getMove() instanceof StatusMove) {
+          movePower = 1;
+        } else if (move.getMove().findAttr(attr => attr instanceof OneHitKOAttr)) {
+          movePower = 150;
+        } else if (move.getMove().id === Moves.COUNTER || move.getMove().id === Moves.MIRROR_COAT || move.getMove().id === Moves.METAL_BURST) {
```
Comment (Xavion3) [inline src/data/ability.ts:1703]: Why isn't this checking for every counter move? Is that actually how forewarn works to ignore one of them?

**Review by Xavion3 — CHANGES_REQUESTED** [review]

File: src/data/ability.ts (hunk 2 above)
Comment (TempsRay) [reply src/data/ability.ts:1703]: According to [Bulbapedia](https://bulbapedia.bulbagarden.net/wiki/Forewarn_(Ability)) and [Smogon](https://www.smogon.com/dex/sv/abilities/forewarn/) Comeuppence is not counted as a counter move for Forewarn. I don't have the cart to test but if anyone can confirm, I can add it in.

Comment (Xavion3) [reply src/data/ability.ts:1703]: Sure, weird but not the weirdest thing we've dealt with.

**Review by Xavion3 — APPROVED** [review]

File: src/data/move.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Xavion3) [reply src/data/move.ts]: This was changed to no longer use a flag

**Pull request merged**